nir/src/main/scala/scala/scalanative/nir/Show.scala

```diff
@@ -33,28 +36,39 @@ object Show {
   def apply(v: Type): String  = { val b = newBuilder; b.type_(v); b.toString }
   def apply(v: Val): String   = { val b = newBuilder; b.val_(v); b.toString }
 
+  type DefnString = (Global, String)
+
   def dump(defns: Seq[Defn], fileName: String): Unit = {
+
+    val collectDefs = new JFunction[(Int, Seq[Defn]), JStream[DefnString]] {
+      override def apply(t: (Int, Seq[Defn])): JStream[DefnString] = {
+        val (_, defns) = t
+        defns
+          .collect {
+            case defn if defn != null =>
+              (defn.name, defn.show)
+          }
+          .asJavaCollection
+          .stream()
+      }
+    }
+
     val pw = new java.io.PrintWriter(fileName)
+    val writeToFile = new JConsumer[DefnString] {
+      override def accept(t: (Global, String)): Unit = {
+        val (_, shown) = t
+        pw.write(shown)
+        pw.write("\n")
+      }
+    }
+
     try {
+      val groupedDefns = util.partitionBy(defns.filter(_ != null))(_.name)
       util
-        .partitionBy(defns.filter(_ != null))(_.name)
-        .par
```

Comment:
Consider just removing `.par` here, instead of going through these hoops of scala-collection-compat + JDK Streams, only to get a parallel `dump` which is not even performance-critical.